zapisy/apps/enrollment/courses/assets/components/CourseList.vue

```diff
@@ -17,22 +17,43 @@ export default Vue.extend({
     }),
   },
   mounted() {
-    // When mounted, load the list of courses from embedded JSON.
+    // When mounted, load the list of courses from embedded JSON and apply initial filters
+    // fetched from the query string.
     const courseData = JSON.parse(
       document.getElementById("courses-data")!.innerHTML
     ) as CourseInfo[];
     this.courses = courseData;
-    this.visibleCourses = courseData;
+    this.visibleCourses = courseData.filter(this.tester);
+
+    // Append the initial query string to links in the semester dropdown.
+    updateSemesterLinks();
 
     this.$store.subscribe((mutation, _) => {
       switch (mutation.type) {
         case "filters/registerFilter":
           this.visibleCourses = this.courses.filter(this.tester);
+          // Update the query string of links in the semester dropdown
+          // to reflect the new state of filters.
+          updateSemesterLinks();
           break;
       }
     });
   },
 });
+
+// Replaces the query string of links in the semester dropdown with the current query string.
+function updateSemesterLinks() {
+  const queryString = window.location.search;
+  const semesterLinks = document.getElementsByClassName("semester-link");
+  for (let i = 0; i < semesterLinks.length; i++) {
+    const link: Element = semesterLinks[i];
```

Comment:
Nit: Czy pętla for-of nie byłaby tutaj odpowiedniejsza

Reply:
Przy próbie bezpośredniego iterowania po `semesterLinks` za pomocą for-of, Vetur zgłasza błąd:

> Type 'HTMLCollectionOf<Element>' must have a '\[Symbol.iterator]()' method that returns an iterator. Vetur(2488)

Należałoby więc utworzyć tablicę z tego obiektu za pomocą `Array.from()`:
```suggestion
  for (const link of Array.from(semesterLinks)) {
```

Nie byłem pewien czy takie rozwiązanie byłoby lepsze od zastosowania zwykłej pętli for.

Reply:
Myślę, że można zostawić jak jest.